server/controllers/middlewares.js

```diff
@@ -54,12 +56,16 @@ module.exports = function(app) {
       }
 
       const userData = {
+        name,
         email: email || paypalEmail,
         paypalEmail
       };
       models.User.findOne({
         where: {
-          $or: userData
+          $or: {
+            email: userData.email,
+            paypalEmail: userData.paypalEmail
+          }
         }
       })
       .then(user => user || users._create(userData))
```

Comment:
could also leave `userData` intact and use `Object.assign({}, userData, {name})`